src/AracariNode.ts

```diff
@@ -0,0 +1,153 @@
+import { Instruction, InstructionType, MinimalNode, NodeType } from "./types";
+import { Aracari } from "./Aracari";
+
+interface AracariNodeOptions {
+  originalNode: MinimalNode;
+  parentNode?: AracariNode;
+  onUpdate?: (instruction: Instruction) => void;
+  root?: Aracari;
+  id?: string;
+}
+
+const noop = () => {};
+
+export class AracariNode {
+  __type = "ARACARI_NODE";
+  nodeType: number;
+  _content?: string;
+  id?: string;
+  childNodes: AracariNode[];
+  parentNode?: AracariNode;
+  originalNode: MinimalNode;
+  onUpdate: (instruction: Instruction) => void;
+  root?: Aracari;
```

Comment:
I think this is the only real circular dependency. I am gonna add eslint and the no-cycle rule to make sure to sus this out.